solr/core/src/java/org/apache/solr/handler/loader/JsonLoader.java

```diff
@@ -664,8 +672,38 @@ private Object parseSingleFieldValue(int ev, String fieldName) throws IOExceptio
         if (ev == JSONParser.ARRAY_END) {
           return lst;
         }
-        Object val = parseSingleFieldValue(ev, fieldName);
+        Object val = parseSingleFieldValue(ev, sif);
         lst.add(val);
+        sif.setValue(null);
+      }
+    }
+
+    private boolean isChildDoc(SolrInputDocument extendedMap) {
+      return extendedMap.containsKey(req.getSchema().getUniqueKeyField().getName());
+    }
+
+    private boolean entryIsChildDoc(Object val) {
+      if(val instanceof List) {
+        List listVal = (List) val;
+        if (listVal.size() == 0) return false;
+        return  listVal.get(0) instanceof Map;
+      }
+      return val instanceof Map;
+    }
+
+    private SolrInputDocument generateExtendedValueDoc(int ev) throws IOException {
```

Comment:
the name seems off -- 'generate' isn't bad but this method is parsing and plenty of other methods here parse and are named as such so I think this should be `parseExtendedValueAsDoc` or better `parseExtendedFieldValueAsDoc` since it is only called by `parseExtendedFieldValue`.